### PR TITLE
[IMP] core: add early return for exists() method based on the field cache

### PR DESCRIPTION
**Pull request by ryv-odoo**
The exists() method is mainly used by env.ref(), to check for missing
records in the ORM, by the fields.Reference.convert_to_record and to
check the existence of a record when the id comes from a route. It
always made a SQL query to the database to do this. Although this query
is very fast, there is still the cost of "latency". We can actually use
the field cache to know if a record is in the database or not. In fact,
because the unlink() clears the entire field cache, we can ensure (if
the cache consistency is correct) that if there is at least one field
value for a particular record in the cache, then that record exists in
the database.

Of course there is a CPU cost to this new early return, it takes about
4.81% of the SQL query time. For the enterprise at_install batch of
runbot, it avoids us the query 41.75% of the time.

For the install community (runbot):
- exists() took +-  1455 ms and now takes +-  1182 ms in total
For the enterprise at_install batch of runbot:
- exists() took +- 29463 ms and now takes +- 18460 ms in total

### CODE REVIEW

**Review by kmagusiak — APPROVED** [review]

File: odoo/models.py [hunk 1]
```diff
@@ -5726,6 +5726,14 @@ def exists(self) -> Self:
         new_ids, ids = partition(lambda i: isinstance(i, NewId), self._ids)
         if not ids:
             return self
+
+        # Special version without query
```
Comment (kmagusiak) [inline odoo/models.py:5730]: More useful comment
```suggestion
        # Cache is invalidated for unlinked records, if all records are in cache, they exist
```

Comment (rco-odoo) [reply odoo/models.py:5730]: I would suggest: "if a record has data in cache, then it exists (thanks to cache consistency)"

File: odoo/models.py [hunk 2]
```diff
@@ -5726,6 +5726,14 @@ def exists(self) -> Self:
...
+        field_to_check = next((f for f in self._fields.values() if f.prefetch is True), None)
+        if field_to_check:
+            field_cache = self.env.cache._get_field_cache(self, field_to_check)
+            if all(_id in field_cache for _id in ids):
```
Comment (rco-odoo) [inline odoo/models.py:5734]: A variable starting with `_` is meant to be unused.
```suggestion
            if all(id_ in field_cache for id_ in ids):
```

Comment (HydrionBurst) [inline odoo/models.py:5734]: I remember when `write` we cannot promise the `record` exist, but just update the field cache.
So `_id` in the `field_cache` here may be caused by `write`.
But maybe it is acceptable here?

**Review by rco-odoo — COMMENTED** [review]
A test is broken: probably some SQL deleting records.  If you can find that code, simply invalidate the deleted records.

File: odoo/models.py (hunk 2 above)
Comment (rco-odoo) [reply odoo/models.py:5734]: That's a good point.

Comment (ryv-odoo) [reply odoo/models.py:5734]: Nice catch, I didn't think about it. Thanks

> But maybe it is acceptable here?

IMO, No it is not. I cancel it.

**Pull request closed without merging**